Automattic/sensei
# Last onboarding step data handling

Part of #5647
Based on https://github.com/Automattic/sensei/pull/5780

### Changes proposed in this Pull Request

* It updates the Setup Wizard data handling (frontend store, and the API) to support submitting data for multiple steps at the same time (when we select the purpose, we also want to define what will be installed later). Before doing it, I thought of a different approach to have a check in the purpose step only to show the special messages when the checkbox is selected, and another check in the last step to install the plugins based on the selected purposes (but it would make less sense in the code, having some kind of logic duplication or weird code).
* It implements the correct messages in the purpose step when selecting the options (it says that it will install or activate the respective plugin if it's not there yet, or no message if it's already activated).
* It deprecates some more code.

### Testing instructions


* Setup a site without WooCommerce and Sensei Certificates installed.
* Navigate to the purpose step, and make sure that when you check the 2 first items, it will display a message that it will install WooCommerce and Sensei Certificates.
* Now, install both plugins, but don't activate them.
* Go to the purpose step, and make sure now you see the messages saying that it will activate the plugins.
* Navigate until the next step, and make sure the plugins to be installed/activated are displayed in a temporary text.


### Deprecated Code

* ~~`Sensei_REST_API_Setup_Wizard_Controller::register_submit_features_route`~~ (reactivated with a new purpose in https://github.com/Automattic/sensei/pull/5824)
* `Sensei_REST_API_Setup_Wizard_Controller::register_submit_features_installation_route`
* `Sensei_REST_API_Setup_Wizard_Controller::register_get_features_route`
* ~~`Sensei_REST_API_Setup_Wizard_Controller::submit_features`~~ (reactivated with a new purpose in https://github.com/Automattic/sensei/pull/5824)
* `Sensei_REST_API_Setup_Wizard_Controller::submit_features_installation`
* `Sensei_Setup_Wizard::install_extensions`


### Screenshot / Video

https://user-images.githubusercontent.com/876340/193116722-083948a8-8c9d-4db4-9363-c82b061c090c.mov
